File: capitalize_words/capitalize_words.rb
# CAPITALIZE WORDS
# CHALLENGE DESCRIPTION:


# Write a program which capitalizes the first letter of each word in a sentence.

# INPUT SAMPLE:

# Your program should accept as its first argument a path to a filename. Input example is the following

# Hello world
# javaScript language
# a letter
# 1st thing
# OUTPUT SAMPLE:

# Print capitalized words in the following way.

# Hello World
# JavaScript Language
# A Letter
# 1st Thing
# SUBMIT SOLUTION
def solution str
  str.split(' ').map do |w|
    letters = w.chars.to_a
    letters.unshift(letters.shift.upcase).join
  end.join ' '
end

def init file
  IO.readlines(file).each { |line| puts solution line.chomp }
end

init ARGV[0]

File: capitalize_words/capitalize_words.js
function solution(line) {
    var characters = line.split('');
    characters[0] = characters[0].toUpperCase();
    for (var i = 0; i < characters.length; i ++) {
        if(characters[i] === ' ') {
            characters[i+1] = characters[i+1].toUpperCase();
        }
    }
    return characters.join('');
}

module.exports.solution = solution;

var fs = require('fs');
function init(file) {
    'use strict';
    fs.readFile(file, 'utf-8', function (err, data) {
        // prep the file data
        if (err) throw err;
        var lines = data.split("\n");
        lines.pop();

        // feed to solution line by line
        lines.forEach(function(line) {
            console.log(solution(line));
        });
    });
}

init(process.argv[2]);